### ISSUE TITLE
Provide a way to override the mapping prefix all at once

### ISSUE THREAD

**Issue by Julian** [nvim-orgmode/orgmode#329]
### Does this feature exist in Emacs orgmode core?

N/A

### Feature value

nvim-orgmode uses `<Leader>o` for lots of its default mappings.

I see there's `disable_all` to disable it from doing so, and that I can explicitly remap all the keys in a similar way in its config, but I'd like to essentially *move* all the mappings to an unused key, rather than explicitly map them. Specifically I'd want all of orgmode's mappings to be on [`<localleader>`](https://neovim.io/doc/user/map.html#%3CLocalLeader%3E) which is more suitable for per-ft mappings.

Would it be reasonable to offer an additional config setting, such as:

```
mappings = {
    prefix = '<Leader>o',
}
```

allowing for overriding the above via `prefix = '<localleader>'`, without needing to repeat all the mappings?

Thanks for the plugin.

**Comment by jgollenz** [comment]
Good idea. Currently `<Leader>o` is hard-coded in the defaults and putting `prefix = <Leader>o` into the same table as the other mappings won't work because Lua tables can't refer to themselves before they are created, e.g:

```lua
mappings = {
    prefix = "<Leader>o",
    org = {
        org_set_tags_command = mappings.prefix .. 't', -- error
    }
}
```

So what we would need to do is indicate somehow, that certain mappings need a prefix and others don't. The easiest way I can think of is just replace all the occurrences of `<Leader>o` with something like `<PREFIX>` and replace that with the value in `mappings.prefix` later on. We could also make something like `org_set_tags_command = { 'true', 't' }`, but then we really need to touch all of the mappings and add info whether it needs a prefix or not.

**Comment by kristijanhusak** [comment]
@jgollenz that sounds ok. You can go around the error by converting `config/defaults.lua` to return a function, and pass a prefix inside if it's provided from the user opts. Inside the function you just use what's provided or default to `<Leader>o`.

**Comment by jgollenz** [comment]
That would mean, that user opts contains something for which there is no default in `config/defaults.lua`. Is that something we already have? Feels a bit weird to me, but my solution is also a hack. The prefix would have to be on a meta level one way or the other :man_shrugging: ran into some problems with my approach, will try yours ;)

**Comment by kristijanhusak** [comment]
We merge the user config with default [here](https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/config/init.lua#L14). You can check if user provided something from opts and pass it down to that function. If not, just default to `<Leader>o`.

**Comment by jgollenz** [comment]
Yeah I know, I just never thought that something will come from the user that is not also reflected as something in the defaults. Its a special case, hence weird :grin: 

Actually it looks like that won't work, because `Config:new` is called without user opts and by the the time we call `Config:extend` we are already extending the opts that we loaded from defaults. The user opt to change the defaults will never reach the defaults this way.
